dask/dataframe/tests/test_multi.py

```diff
@@ -1627,38 +1627,35 @@ def fix_index(out, dtype):
             )
 
 
-@pytest.mark.xfail_with_pyarrow_strings  # https://github.com/dask/dask/issues/10029
-def test_melt():
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {},
+        dict(id_vars="int"),
+        dict(value_vars="int"),
+        dict(value_vars=["obj", "int"], var_name="myvar"),
+        dict(id_vars="s1", value_vars=["obj", "int"], value_name="myval"),
+        dict(value_vars=["obj", "s1"]),
+        dict(value_vars=["s1", "s2"]),
+    ],
+)
+def test_melt(kwargs):
     pdf = pd.DataFrame(
-        {"A": list("abcd") * 5, "B": list("XY") * 10, "C": np.random.randn(20)}
+        {
+            "obj": list("abcd") * 5,
+            "s1": list("XY") * 10,
+            "s2": list("abcde") * 4,
+            "int": np.random.randn(20),
+        }
     )
-    ddf = dd.from_pandas(pdf, 4)
+    pdf.s1 = pdf.s1.astype(pd.StringDtype("pyarrow"))
+    pdf.s2 = pdf.s2.astype(pd.StringDtype("pyarrow"))
```

Comment:
Is this intended, or left over from local development?

Reply:
This is intended, I want `pandas` to melt together some `pyarrow` strings.

Reply:
Ah, I missed that these are new columns you're adding. In that case, this looks fine. I would add some additional logic to handle the case that `pyarrow` isn't installed and maybe simplify to 

```python
pdf = pdf.astype({"s1": "string[pyarrow]", "s2": "string[pyarrow]"})
```

Reply:
I went ahead and pushed a tiny commit that makes sure `test_melt` is run, even when `pyarrow` isn't installed. Hope that's okay